tests/resources/test_dashboard.py

```diff
@@ -29,3 +52,364 @@ def test_dashboard(self):
+
+    @pytest.mark.skipif(
+        os.environ.get("CI_JIRA_TYPE", "Server").upper() != "CLOUD",
+        reason="Functionality only available on Jira Cloud",
+    )
+    @allow_on_cloud
+    def test_create_dashboard(self):
+        name = rndstr()
+        description = rndstr()
+        share_permissions = [{"type": "authenticated"}]
+
+        dashboard = self.jira.create_dashboard(
+            name=name, description=description, share_permissions=share_permissions
+        )
+        self.dashboards_to_delete.append(dashboard)
+
+        self.assertEqual(dashboard.name, name)
+        self.assertEqual(dashboard.description, description)
+        # NOTE(jpavlav): This is a bit obtuse, but Jira mutates the type on this
+        # object after the fact. `authenticated` corresponds to `loggedin`.
+        self.assertEqual(dashboard.sharePermissions[0].type, "loggedin")
+
+        # NOTE(jpavlav): The system dashboard always has the ID `10000`, just
+        # ensuring we actually have a
+        self.assertGreater(int(dashboard.id), 10000)
+
+    @pytest.mark.skipif(
+        os.environ.get("CI_JIRA_TYPE", "Server").upper() != "CLOUD",
+        reason="Functionality only available on Jira Cloud",
+    )
+    @allow_on_cloud
+    def test_update_dashboard(self):
+        updated_name = "changed"
+        name = rndstr()
+        description = rndstr()
+        share_permissions = [{"type": "authenticated"}]
+
+        dashboard = self.jira.create_dashboard(
+            name=name, description=description, share_permissions=share_permissions
+        )
+        self.dashboards_to_delete.append(dashboard)
+
+        dashboard.update(name=updated_name)
+        self.assertEqual(dashboard.name, updated_name)
+
+    @pytest.mark.skipif(
+        os.environ.get("CI_JIRA_TYPE", "Server").upper() != "CLOUD",
+        reason="Functionality only available on Jira Cloud",
+    )
+    @allow_on_cloud
+    def test_delete_dashboard(self):
+        dashboard = self.jira.create_dashboard(name="to_delete")
+        dashboard_id = dashboard.id
+        delete_response = dashboard.delete()
+        self.assertEqual(delete_response.status_code, 204)
+
+        with pytest.raises(JIRAError) as ex:
+            self.jira.dashboard(dashboard_id)
+
+        self.assertEqual(ex.value.status_code, 404)
+        self.assertEqual(
+            ex.value.text, f"The dashboard with id '{dashboard_id}' does not exist."
+        )
+
+    @pytest.mark.skipif(
+        os.environ.get("CI_JIRA_TYPE", "Server").upper() != "CLOUD",
+        reason="Functionality only available on Jira Cloud",
+    )
+    @allow_on_cloud
+    def test_copy_dashboard(self):
+        original_dashboard = self.jira.create_dashboard(
+            name=rndstr(), share_permissions=[{"type": "authenticated"}]
+        )
+        self.dashboards_to_delete.append(original_dashboard)
+        # NOTE(jpavlav): Add something to the dashboard so we can test the copy worked
```

Comment:
As these comments are actually describing the test flow, it is better to use the

Given, When, Then

Syntax for the test comments.

Typically you associate your name with TODOs more than notes like this, as the git history can always tell us who specifically.
There are quite a few instances throughout this PR and it would be good to identify which should be turned into normal comments vs. which were your notes while you were developing. The latter which we should not be leaving in the code